addons/mass_mailing/models/mass_mailing.py

```diff
@@ -578,7 +578,25 @@ def _get_blacklist(self):
         blacklist = {}
         target = self.env[self.mailing_model_real]
         mail_field = 'email' if 'email' in target._fields else 'email_from'
-        if 'opt_out' in target._fields:
+        if self.mailing_model_real == "mail.mass_mailing.contact":
+            # If user is opt_out on One list but not on another or if two user
+            # with same email address, one opted in and the other one
+            # opted out, send the mail anyway
+            query = """
+                SELECT lower(substring(email, '([^ ,;<@]+@[^> ,;]+)')), opt_out
+                FROM mail_mass_mailing_contact
+                WHERE id IN (
```

Comment:
Do a join if possible (AFAICS it is), to avoid problems like https://github.com/odoo/odoo/issues/30350.